### PR TITLE
Universal listings: Add 'pills' to show active filters

### PR DESCRIPTION
**Pull request by gasman**
Display the currently active filters, along with links to remove them individually.

Screenshot 2024-01-11 at 17 14 06

Marking as draft for now, since we probably don't want to merge this until we have the 'pill' styling in place.

### CODE REVIEW

File: wagtail/admin/views/generic/base.py [hunk 1]
```diff
@@ -199,6 +214,96 @@ def filter_queryset(self, queryset):
...
+        Return the index URL where the filter parameter with the given value has been removed
+        from the query string, preserving all other values for that parameter
+        """
+        base_url = self.index_url.split("?")[0]
+        query_dict = self.request.GET.copy()
+        query_dict.pop(self.page_kwarg, None)  # reset pagination to first page
+        query_dict.setlist(
+            param, [v for v in query_dict.getlist(param) if v != str(value)]
+        )
+        return base_url + "?" + query_dict.urlencode()
```
Comment (laymonage) [inline wagtail/admin/views/generic/base.py:242]: Drive-by question: would the `set_query_param` util be of any use here?

https://github.com/wagtail/wagtail/blob/0ee97d45deacae6e210207e3ebdd3b0b94afb4c7/wagtail/admin/utils.py#L59-L72

Comment (gasman) [reply wagtail/admin/views/generic/base.py:242]: It'll do the job for `get_url_without_filter_param` (but not `get_url_without_filter_param_value`), but I'd rather avoid the overhead of parsing a URL when we already have a QueryDict.

**Review by laymonage — APPROVED** [review]
Looks good for the most part! I think the `bubble` addition should be applied to the other event handlers in date-time-chooser.js as well. I made a separate PR (#11446) with this fix as well as the stylings and AJAX.

I think we should also ensure that the `BaseListingView` template works with the filters, but that's something we can handle in a separate PR.

File: wagtail/admin/templates/wagtailadmin/generic/index_results.html [hunk 2]
```diff
@@ -2,6 +2,10 @@
 {% load i18n wagtailadmin_tags %}
 
 {% block before_results %}
+    {% if view.active_filters %}
+        {% include "wagtailadmin/shared/active_filters.html" with active_filters=view.active_filters %}
+    {% endif %}
+
```
[the remark this answers is not in the document]
Comment (gasman) [reply wagtail/admin/templates/wagtailadmin/generic/index_results.html:8]: I figured we _didn't_ want to include the active filters block on listings that use the old-style page furniture (such as reports) where the filter form is always visible, and rather than go through the 'is breadcrumb_items' dance to decide whether that's the case or not, I thought it would be cleaner to add it individually as and when those views get ported over.

**Comment by thibaudcolas** [comment]
Merged in #11452
